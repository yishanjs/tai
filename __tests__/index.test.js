/**
 * Copyright (c) 2019 Yishan Authors
 *
 * All right reserved
 */

import * as React from 'react';
import { shallow } from 'enzyme';

import App from '@/pages/index.tsx';

describe('Index', () => {
  it('should shows"Yishanjs\'s typescript&next starter"', async () => {
    const props = await App.getInitialProps();
    const app = shallow(<App {...props} />);

    expect(app.find('h1').text()).toEqual("Yishanjs's typescript&next starter");
  });

  it('should match snapshot', async () => {
    const props = await App.getInitialProps();
    const app = shallow(<App {...props} />);

    expect(app).toMatchSnapshot();
  });
});
