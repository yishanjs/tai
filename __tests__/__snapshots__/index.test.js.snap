// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index should match snapshot 1`] = `
<LandingLayout>
  <div>
    <h1>
      Yishanjs's typescript&next starter
    </h1>
    <ul>
      <li
        key="nextjs"
      >
        <Link
          href="https://github.com/zeit/next.js/"
          passHref={true}
        >
          <a
            target="_blank"
          >
            nextjs
          </a>
        </Link>
      </li>
      <li
        key="next-typescript"
      >
        <Link
          href="https://github.com/zeit/next-plugins/tree/master/packages/next-typescript"
          passHref={true}
        >
          <a
            target="_blank"
          >
            next-typescript
          </a>
        </Link>
      </li>
      <li
        key="typescript"
      >
        <Link
          href="https://www.typescriptlang.org/"
          passHref={true}
        >
          <a
            target="_blank"
          >
            typescript
          </a>
        </Link>
      </li>
      <li
        key="eslint"
      >
        <Link
          href="https://eslint.org/"
          passHref={true}
        >
          <a
            target="_blank"
          >
            eslint
          </a>
        </Link>
      </li>
      <li
        key="prettier"
      >
        <Link
          href="https://github.com/prettier/prettier"
          passHref={true}
        >
          <a
            target="_blank"
          >
            prettier
          </a>
        </Link>
      </li>
      <li
        key="jest"
      >
        <Link
          href="https://github.com/prettier/prettier"
          passHref={true}
        >
          <a
            target="_blank"
          >
            jest
          </a>
        </Link>
      </li>
    </ul>
  </div>
</LandingLayout>
`;
